py4web/utils/grid.py

```diff
@@ -668,6 +672,17 @@ def render_table_header(self):
         dw = I(**self.param.grid_class_style.get("grid-sorter-icon-down"))
         columns = []
         sort_order = request.query.get("orderby", "")
+        if sort_order:
+            sort_order = [sort_order]
+        else:
+            sort_order = []
+            for x in self.param.orderby:
+                op = ''
+                if hasattr(x, 'op') and x.op and x.op.__name__ == 'invert':
+                    x = x.first
+                    op = '~'
+                sort_order.append("%s%s.%s" % (op, x.tablename, x.name))
```

Comment:
Why append and not inert(0,...)?

Reply:
I don't have a good reason.  This section of code something (I think) Val had suggested when I had a problem earlier with recognizing DESC.  I have no problem if you'd like to change it.

Reply:
i will merge everything but this because unfortunately does not work as it unless we restrict orderby to expressions that are representable as strings. Need to think more about it. Notice the user can always do: if not request.query.get('orderby'): sequet.query['orderby'] = "~tablename.fieldname".

Reply:
Sounds good.  I will try to find a better solution.